src/dali/plugin/input/rest/coinbase.py

```diff
@@ -339,13 +339,15 @@ def _process_fill(
             self.__logger.warning("Swap transaction encountered (swaps not supported yet by Coinbase plugin): %s", json.dumps(transaction))
             return
 
-        if transaction_type in {_BUY, _TRADE}:
-            spot_price = (native_amount - fiat_fee) / crypto_amount
+        if transaction_type == _BUY or (transaction_type == _TRADE and native_amount >= ZERO):
             if transaction_type == _BUY:
                 buy: Dict[str, Any] = id_2_buy[transaction[transaction_type][_ID]]
                 fiat_fee = RP2Decimal(buy[_FEE][_AMOUNT])
                 self.__logger.debug("Buy: %s", json.dumps(buy))
                 spot_price = RP2Decimal(buy[_UNIT_PRICE][_AMOUNT])
+            else:
+                # swap in transaction
+                spot_price = (native_amount - fiat_fee) / crypto_amount
```

Comment:
In this path fiat_fee is always 0: this may be the correct for your specific case (is Coinbase not charging fee when swapping ETH -> ETH2?), but I suspect when swapping coins there will be a fee and it will be probably paid in crypto (similar to what happens on Coinbase Pro). Do you have another example of coin swap handy other than ETH -> ETH2? If not we can follow-up on issue #15: @jameskupke has posted the Buy half of an Algo -> XTZ conversion. We need to get the Sell side as well. 

Note that in RP2 passing a crypto_fee to an InTransaction is not the equivalent to passing a fiat_fee. If fiat_fee is assigned, crypto_fee is set to 0 by RP2. If crypto_fee is assigned, fiat_fee is set to crypto_fee * spot_price by RP2. The reason for this behavior is that if the fee is paid in fiat, then no crypto is used for the fee, but if the fee is paid in crypto, then its converted fiat value is needed to compute taxes. Note that RP2 models a non-zero crypto_fee with a separate fee-typed out-transaction.

In your case was there a fee and if so was it crypto or fiat?

Reply:
Also if there was a crypto fee was it paid in ETH or ETH2?

Reply:
To summarize my position: I think this PR is good to start with (it does solve the negative amount problem), but down the road we may have to modify the _TRADE branches to account for crypto fees.

Reply:
There is no fee based on the api response when you stake ETH for ETH2. The PR only focuses on ETH2 stake because that's the only swap transaction I have in coinbase :) I am pretty sure more work is needed to handle other type of swaps.

Reply:
Sounds good, let's go with what we have and we'll keep an eye out for crypto fees later.